release-notes.txt

```diff
@@ -1,3 +1,565 @@
+found in the Internet of Things: 8-bit and 16-bit microcontrollers as well as
+light-weight 32-bit processors.
+
+RIOT is based on the following design principles: energy-efficiency, soft
+real-time capabilities, small memory footprint, modularity, and uniform API
+access, independent of the underlying hardware (with partial POSIX compliance).
+
+RIOT is developed by an international open-source community which is
+independent of specific vendors (e.g. similarly to the Linux community) and is
+licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
+business models around the free open-source software platform provided by RIOT.
+
+
+About this release
+==================
+
+The 2025.04 release includes:
+
+RIOT includes `native` as a board that can be used for fast development cycles
+on the development machine without the need for external microcontroller hardware.
+Starting with this release, `native` will default to a 64-bit executable on 64-bit
+hosts (#21242). If you prefer to cross-compile 32-bit executables, please now use
+`native32` instead.
+
+This new board alias mechanism also paves the way towards more consistent board
+names in RIOT. As a first step, the two Adafruit Feather nRF52840 boards have
+been renamed, while maintaining the old board names as aliases (#21349).
+
+The examples contained in the RIOT source repository have been restructured as
+a first step towards easing the experience for RIOT newcomers (#21135, #21221).
+Stay tuned for more!
+
+This release includes support for three new boards from the Seeed Studio XIAO
+family, notably the nRF52840 (#20980), nRF52840 Sense (#21332), and the ESP32C3
+variant (#21267). RIOT now also boots on SAM4S Xplained Pro evaluation kits (#21201).
+
+175 pull requests, composed of 357 commits, have been merged since the
+last release, and 12 issues have been solved. 35 people contributed with
+code in 82 days. 1963 files have been touched with 49418 (+) insertions and
+37661 deletions (-).
+
+
+Notations used below
+====================
+
+    + means new feature/item
+    * means modified feature/item
+    - means removed feature/item
+
+
+New features and changes
+========================
+
+Core
+----
+
++ core/cib: Add a prominent warning about thread safety (#21401)
+* core/debug: extend doc about DEBUG() / LOG_DEBUG() (#21276)
+* core/thread: error in thread.c corrected (#21253)
+
+System Libraries
+----------------
+
++ sys/bcd: add bcd_buf_from_u32() (#21308)
++ sys/event: add `event_deferred_callback_post()` helper (#21219)
++ sys/event: add `event_loop_debug` pseudo-module (#20849)
++ sys/fmt: add {fmt,scn}_time_tm_iso8601 (#21391)
++ sys/psa_crypto: Adding hmac hashing on psa_import_key and fix max
+  hmac key size (#21297)
++ sys/ztimer: add hint for setting static sleep adjustment (#21376)
+* drivers/mtd_default: remove deprecated `mtd_default_get_dev()` (#21386)
+* Fido2 update tests (#20876)
```

Comment:
```suggestion
* sys/fido2: update tests (#20876)
```

I was so free to also update the title in https://github.com/RIOT-OS/RIOT/pull/20876 to match